marlowe/src/Language/Marlowe/Client.hs

```diff
@@ -0,0 +1,206 @@
+import           Control.Monad.Error.Class  (MonadError (..))
+import           Data.Map                   (Map)
+import qualified Data.Map                   as Map
+import           Data.Maybe                 (maybeToList)
+import qualified Data.Set                   as Set
+import qualified Data.Text                  as Text
+import           Language.Marlowe.Semantics as Marlowe
+import qualified Language.PlutusTx          as PlutusTx
+import           Ledger                     (DataScript (..), PubKey (..), Slot (..), Tx, TxOut, TxOutOf (..), interval,
+                                             mkValidatorScript, pubKeyTxOut, scriptAddress, scriptTxIn, scriptTxOut,
+                                             txOutRefs)
+import           Ledger.Ada                 (Ada)
+import qualified Ledger.Ada                 as Ada
+import           Ledger.Scripts             (RedeemerScript (..), ValidatorScript)
+import qualified Ledger.Typed.Scripts       as Scripts
+import           Wallet                     (WalletAPI (..), WalletAPIError, createPaymentWithChange, createTxAndSubmit,
+                                             throwOtherError)
+
+{-| Create a Marlowe contract.
+    Uses wallet public key to generate a unique script address.
+ -}
+createContract :: (
+    MonadError WalletAPIError m,
+    WalletAPI m)
+    => Contract
+    -> m MarloweData
+createContract contract = do
+    slot <- slot
+    creator <- ownPubKey
+    let validator = validatorScript creator
+
+        marloweData = MarloweData {
+            marloweCreator = creator,
+            marloweContract = contract,
+            marloweState = emptyState slot }
+        ds = DataScript $ PlutusTx.toData marloweData
+
+        deposit = Ada.adaValueOf 1
+
+    (payment, change) <- createPaymentWithChange deposit
+    let o = scriptTxOut deposit validator ds
+        slotRange = interval slot (slot + 10)
+        outputs = o : maybeToList change
+
+    void $ createTxAndSubmit slotRange payment outputs
+    return marloweData
+
+
+{-| Deposit 'amount' of money to 'accountId' to a Marlowe contract
+    from 'tx' with 'MarloweData' data script.
+ -}
+deposit :: (
+    MonadError WalletAPIError m,
+    WalletAPI m)
+    => Tx
+    -> MarloweData
+    -> AccountId
+    -> Ada
+    -> m MarloweData
+deposit tx marloweData accountId amount = do
+    pubKey <- ownPubKey
+    applyInputs tx marloweData [IDeposit accountId pubKey amount]
+
+
+{-| Notify a contract -}
+notify :: (
+    MonadError WalletAPIError m,
+    WalletAPI m)
+    => Tx
+    -> MarloweData
+    -> m MarloweData
+notify tx marloweData = applyInputs tx marloweData [INotify]
+
+
+{-| Make a 'choice' identified as 'choiceId'. -}
+makeChoice :: (
+    MonadError WalletAPIError m,
+    WalletAPI m)
+    => Tx
+    -> MarloweData
+    -> ChoiceId
+    -> Integer
+    -> m MarloweData
+makeChoice tx marloweData choiceId choice = applyInputs tx marloweData [IChoice choiceId choice]
+
+
```

Comment:
It's expected to call `applyInputs` if you want to do multiple inputs in a single transaction, otherwise users could use single input helpers: `deposit`, `makeChoice`, `notify`.

Reply:
I mean, you can probably get around without multiple input transactions. But even if you go one-by-one all the time, there will be moments when then only step forward will be to send an empty transaction. So if we want a complete one-by-one interface we will need a helper for empty transaction too, that is why I am suggesting it. As long as `applyInputs` is usable (from Plutus Playgrounds for example) that should not be a problem.